test/integration/cluster_filter_integration_test.cc

```diff
@@ -71,13 +91,69 @@ class PoliteFilterConfigFactory
   }
 
   std::string name() const override { return "envoy.upstream.polite"; }
+
+private:
+  TestParent& test_parent_;
 };
 
-class ClusterFilterIntegrationTest : public testing::TestWithParam<Network::Address::IpVersion>,
-                                     public BaseIntegrationTest {
+std::string ipInitializeUpstreamFiltersTestParamsToString(
+    const testing::TestParamInfo<std::tuple<Network::Address::IpVersion, bool>>& params) {
+  return fmt::format(
+      "{}_{}",
+      TestUtility::ipTestParamsToString(
+          testing::TestParamInfo<Network::Address::IpVersion>(std::get<0>(params.param), 0)),
+      std::get<1>(params.param) ? "do_initialize_upstream_filters"
+                                : "dont_initialize_upstream_filters");
+}
+
+class ClusterFilterIntegrationTestBase
+    : public testing::TestWithParam<std::tuple<Network::Address::IpVersion, bool>>,
+      public TestParent {
 public:
-  ClusterFilterIntegrationTest()
-      : BaseIntegrationTest(GetParam(), ConfigHelper::tcpProxyConfig()), registration_(factory_) {}
+  ClusterFilterIntegrationTestBase() : factory_(*this), registration_(factory_) {
+    Runtime::maybeSetRuntimeGuard("envoy.reloadable_features.initialize_upstream_filters",
+                                  std::get<1>(GetParam()));
+#if 1
```

Comment:
What is this for?

Reply:
I saw some flakiness in the CI, but mostly on other tests, added some debugging in the last commit that I'll remove before this gets merged.